conanfile.py

```diff
@@ -12,7 +20,20 @@ class ffnvcodecConan(ConanFile):
     no_copy_source = True
 
     def source(self):
-        self.run("git clone -b modules_merged --single-branch --depth 1 %s" % (self.repo_url))
+        _target_version, self.repo_branch, self.target_commit = parse_version(self.version)
+        git = tools.Git()
+        git.clone(
+            self.repo_url,
+            branch=self.repo_branch,
+            shallow=True,
+        )
+        current_commit = git.get_commit()
+        if not current_commit.startswith(self.target_commit):
```

Comment:
this would be much simpler if you just use the commit. in https://docs.conan.io/en/latest/reference/tools.html#tools-git it says if you specify shallow then branch may specify any valid git reference - e.g. branch name, tag name, sha256 of the revision.